bait-files/src/test/java/com/google/sps/UserTest.java

```diff
@@ -0,0 +1,131 @@
+        assertTrue(stringWriter.toString().contains("false"));
+  }
+
+  // Username field is null, so return that the username is taken
+  @Test
+  public void nullUsernameTest() throws IOException {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
+        HttpServletResponse response = mock(HttpServletResponse.class);
+
+        when(request.getParameter("username")).thenReturn(null);
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+
+        when(response.getWriter()).thenReturn(writer);
+        new UsernameTakenServlet().doGet(request, response);
+
+        verify(request, atLeast(1)).getParameter("username"); // only if you want to verify username was called...
+        writer.flush(); // it may not have been flushed yet...
+        assertTrue(stringWriter.toString().contains("true"));
+  }
+
+
+  // Username field is an empty string, so return that the username is taken
+  @Test
+  public void emptyStringUsernameTest() throws IOException {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
+        HttpServletResponse response = mock(HttpServletResponse.class);
+
+        when(request.getParameter("username")).thenReturn("");
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+
+        when(response.getWriter()).thenReturn(writer);
+        new UsernameTakenServlet().doGet(request, response);
+
+        verify(request, atLeast(1)).getParameter("username"); // only if you want to verify username was called...
+        writer.flush(); // it may not have been flushed yet...
+        assertTrue(stringWriter.toString().contains("true"));
+  }
+
+  // Add a user and ensure there is a new entry in database
+  @Test
+  public void addUser() throws IOException, ServletException {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
+        HttpServletResponse response = mock(HttpServletResponse.class);
+
+        UserService userService = UserServiceFactory.getUserService();
+
+        System.out.println(userService.getCurrentUser());
+        System.out.println("==================");
+
+        when(request.getParameter(USERNAME_ENTITY_PROPERTY)).thenReturn("Drew8521");
+        when(request.getParameter(FIRST_NAME_ENTITY_PROPERTY)).thenReturn("Andrew");
+        when(request.getParameter(FIRST_NAME_ENTITY_PROPERTY)).thenReturn("Seaman");
+
+        new LoginServlet().doPost(request, response);
+
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        assertEquals(1, ds.prepare(new Query(USER_ENTITY)).countEntities(withLimit(10)));
+  }
+
+    // Add a user and ensure there is a new entry in database
+  @Test
+  public void addUser2() throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    assertTrue(userService.isUserAdmin());
+  }
```

Comment:
It looks like all you are testing here is that Google's UserService class works as intended, so I'd say you don't need this test.  I assume Google has already tested this class themselves =)

Reply:
Removed